feat: unify focus styles with new standard

Francois-Esquire opened the pull request:
## Description

Unifies focus ring across CKEditor with the new focus ring state.

## Detail

[The  global variable `--ck-focus-outer-shadow`](https://github.com/ckeditor/ckeditor5/blob/master/packages/ckeditor5-theme-lark/theme/ckeditor5-ui/globals/_focus.css#L15) has been introduced to the `:root` variables to provide elements with a unified focus ring across the editor. The `--ck-inner-shadow` has been updated to the new focus ring state with inset styling.

`@zendeskgarden/react-theming` was updated to bring the latest updates with the `xs` shadow width (`$ZD-SHADOWWIDTHS-XS`).

![Screenshot 2023-05-19 at 3 57 00 PM](https://github.com/zendeskgarden/ckeditor/assets/12474067/633920a8-e7b6-4410-8955-e81fe916b6cf)

![Screenshot 2023-05-19 at 3 57 38 PM](https://github.com/zendeskgarden/ckeditor/assets/12474067/708d3959-653e-4090-be83-22f61fdb4d16)

![Screenshot 2023-05-19 at 3 59 19 PM](https://github.com/zendeskgarden/ckeditor/assets/12474067/6e313e10-8308-4b84-9f95-244605837e17)

geotrev reviewed (approved):
LGTM

lucijanblagonic commented:
The implementation looks good, but I found a bug with the original implementation of the focus outline in Toolbar. Could we please adjust the focus outline size (blue shadow) from `3px` to `2px` so there is no overlap of focus over neighboring buttons.

![image](https://github.com/zendeskgarden/ckeditor/assets/773721/bd0ff834-635c-4dd0-8901-36919eb7280c)
![image](https://github.com/zendeskgarden/ckeditor/assets/773721/b697304b-632a-4134-9901-c21972929c66)

This will make the focus outlines on the Toolbar the same as we have it on the `Tags`(`2px` instead of `3px`).

--- 

Unrelated, could we also tackle this color of the item when the dropdown is opened.
![image](https://github.com/zendeskgarden/ckeditor/assets/773721/4cc253ce-af68-45fb-8af1-9994d67cc7d8)
It should be `blue-800`.
![image](https://github.com/zendeskgarden/ckeditor/assets/773721/ffd8535d-7c94-41fe-a844-30676cf80a12)

Francois-Esquire commented:
> The implementation looks good, but I found a bug with the original implementation of the focus outline in Toolbar. Could we please adjust the focus outline size (blue shadow) from `3px` to `2px` so there is no overlap of focus over neighboring buttons.
> 
> ![image](https://user-images.githubusercontent.com/773721/239884808-bd0ff834-635c-4dd0-8901-36919eb7280c.png) ![image](https://user-images.githubusercontent.com/773721/239884967-b697304b-632a-4134-9901-c21972929c66.png)
> 
> This will make the focus outlines on the Toolbar the same as we have it on the `Tags`(`2px` instead of `3px`).
> 
> Unrelated, could we also tackle this color of the item when the dropdown is opened. ![image](https://user-images.githubusercontent.com/773721/239885499-4cc253ce-af68-45fb-8af1-9994d67cc7d8.png) It should be `blue-800`. ![image](https://user-images.githubusercontent.com/773721/239885718-ffd8535d-7c94-41fe-a844-30676cf80a12.png)

I've corrected the focus outline size for buttons and updated the "on" text color of buttons to `blue-800`. I did notice something with disabled buttons:
![Screenshot 2023-05-22 at 3 26 40 PM](https://github.com/zendeskgarden/ckeditor/assets/12474067/8e6d24a9-52e8-405b-8373-1349715f0358)
It's getting a focus ring when the button is disabled in the menu.

lucijanblagonic reviewed (approved):
LGTM!

jzempel commented on src/theme/_variables.css:
```diff
@@ -45,8 +46,16 @@
+  --ck-inner-shadow:
+    inset var(--ck-color-base-background) 0 0 0 $ZD-SHADOWWIDTHS-XS,
+    inset var(--ck-color-focus-border) 0 0 0 $ZD-SHADOWWIDTHS-MD;
+  --ck-focus-outer-shadow:
+    var(--ck-color-base-background) 0 0 0 $ZD-SHADOWWIDTHS-XS,
+    var(--ck-color-focus-outer-shadow) 0 0 0 $ZD-SHADOWWIDTHS-SM;
+  --ck-focus-disabled-outer-shadow:
+    var(--ck-color-base-background) 0 0 0 $ZD-SHADOWWIDTHS-XS,
+    var(--ck-color-focus-disabled-shadow) 0 0 0 $ZD-SHADOWWIDTHS-SM;
```
Why was the total shadow width changed from the previous 3px? The focus for toolbar buttons is now inconsistent with Garden icon buttons.

jzempel replied:
> so there is no overlap of focus over neighboring buttons

@lucijanblagonic the overlap here is intended by the original designs; it was deemed more important for these buttons to mimic Garden icon buttons after which they are patterned. I feel like that intent still holds.

lucijanblagonic replied:
@jzempel I am OK with the decision to revert if we want to aim at consistency and we don't think overlap is much of an issue.

Francois-Esquire replied:
updated in [cc62a90](https://github.com/zendeskgarden/ckeditor/pull/90/commits/cc62a901fe50cc2d3ace7e842cf66a09163b8fbb)!

jzempel reviewed (changes requested):
Focus for the horizontal line is missing the inner box-shadow spacer.
